### PR TITLE
Add Indicators/fdi

### PR DESCRIPTION
**Pull request by Tzumx**

### CODE REVIEW

File: qube/quantitative/ta/indicators.py [hunk 1]
```diff
@@ -1447,3 +1448,27 @@ def psar(ohlc, iaf=0.02, maxaf=0.2):
...
+        priceMin = x[pos:pos+e_period].min()
+        length = 0
+        priorDiff = 0
+        
+        for iteration in range(e_period - 1):
+            diff = (x.iloc[pos + iteration] - priceMin) / (priceMax - priceMin)
+            if iteration > 0:
+                length += np.power(np.power((diff - priorDiff), 2.0) + (1.0 / math.pow(e_period, 2.0)), 0.5)
+                priorDiff = diff
+        fdi = 1.0 + (np.log(length) + math.log(2.0)) / math.log(2*e_period)
```
Comment (dmarienko) [inline qube/quantitative/ta/indicators.py]: I'd better use np.log here instead of math.

File: qube/quantitative/ta/indicators.py [hunk 2]
```diff
@@ -1447,3 +1448,27 @@ def psar(ohlc, iaf=0.02, maxaf=0.2):
     psar_i, psarbear, psarbull = __psar(ohlc['close'].values, ohlc['high'].values, ohlc['low'].values, iaf, maxaf)
 
     return pd.DataFrame({"psar": psar_i, "up": psarbear, "down": psarbull}, index=ohlc.index)
+
+def fdi(x, e_period = 30):
+    fdi_result = pd.DataFrame()
+    
+    lastBars = x.count().min() - e_period
```
Comment (dmarienko) [inline qube/quantitative/ta/indicators.py]: Don't we need to check if x is instance of Series before calling methods like .count() etc ?
What if x is np.array of just a list of values ?

File: qube/quantitative/ta/indicators.py [hunk 3]
```diff
@@ -1447,3 +1448,27 @@ def psar(ohlc, iaf=0.02, maxaf=0.2):
     psar_i, psarbear, psarbull = __psar(ohlc['close'].values, ohlc['high'].values, ohlc['low'].values, iaf, maxaf)
 
     return pd.DataFrame({"psar": psar_i, "up": psarbear, "down": psarbull}, index=ohlc.index)
+
+def fdi(x, e_period = 30):
```
Comment (dmarienko) [inline qube/quantitative/ta/indicators.py]: It'd be better to add types annotations here, x : Union[Series, .....] etc.

**Review by dmarienko — CHANGES_REQUESTED** [review]

File: qube/quantitative/ta/indicators.py (hunk 1 above)
Comment (Tzumx) [reply qube/quantitative/ta/indicators.py]: Done

File: qube/quantitative/ta/indicators.py (hunk 2 above)
Comment (Tzumx) [reply qube/quantitative/ta/indicators.py]: refactoring at all

File: qube/quantitative/ta/indicators.py (hunk 3 above)
Comment (Tzumx) [reply qube/quantitative/ta/indicators.py]: Done

Comment (Tzumx) [reply qube/quantitative/ta/indicators.py]: Done

**Review by dmarienko — APPROVED** [review]

**Pull request merged**